src/main/java/com/snowflake/kafka/connector/records/SnowflakeRecordContent.java

```diff
@@ -109,4 +124,20 @@ int getSchemaID() {
     assert content != null;
     return content.clone();
   }
+
+  /**
+   * Check if primary reason for this record content's value to be an empty json String, a null
+   * value?
+   *
+   * <p>i.e if value passed in by record is empty json node (`{}`), we don't interpret this as null
+   * value.
+   *
+   * @return true if content value is empty json node as well as isNullValueRecord is set to true.
+   */
+  public boolean isRecordContentValueNull() {
+    if (content != null && content[0].isEmpty() && isNullValueRecord) {
```

Comment:
Under what scenario is content null? Is it added just for safety?
Also for my understanding, does a null record json look like this
`[
    null
]`

Reply:
I added content != null and content[0].isEmpty for sanity check. 
A null record looks like this `{}`
(Check constructor for SnowflakeRecordContent)
```
public SnowflakeRecordContent() {
    content = new JsonNode[1];
    content[0] = MAPPER.createObjectNode();
    brokenData = null;
  }

Reply:
Hmm, but isn't content an array? So does it look like this then?
`[ {} ]`

Reply:
That is correct. Looks like we go through the array and create the content. (Meta and actual content)
https://github.com/snowflakedb/snowflake-kafka-connector/blob/master/src/main/java/com/snowflake/kafka/connector/records/RecordService.java#L129-L139

Reply:
So ultimately this is what it looks like: 
{"content":{},"meta":{"topic":"test","offset":0,"partition":0,"schema_id":0,"key":"test"}}